### DISCUSSION TITLE
大佬是否可以提供chatgpt-proxy-server和chatgpt-proxy-server-warp编译好的文件

### DISCUSSION THREAD

**Discussion by kakamatong** [linweiyuan/go-chatgpt-api#54] [General]
代理机器，配置很低，搭建docker环境不现实，希望提供编译好的文件，我直接部署

**Comment by linweiyuan** [comment]
proxy server 这个可以给，但只有 Linux 版本

warp 这个可以下 Cloudflare 官方客户端

**Reply by kakamatong** [reply]
可以的，多谢了

**Reply by linweiyuan** [reply]
供参考

```bash
➜  ~ docker run -d --rm linweiyuan/chatgpt-proxy-server
1a2216db8e02cea50e287b138b72368bfe9f3749e02486d5b0c27179eba97e1b
➜  ~ docker cp 1a2:/undetected_chromedriver .
Successfully copied 13.98MB to /home/linweiyuan/.
➜  ~ ./undetected_chromedriver 
Starting ChromeDriver 111.0.5563.64 (c710e93d5b63b7095afe8c2c17df34408078439d-refs/branch-heads/5563@{#995}) on port 9515
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
```

warp：https://developers.cloudflare.com/warp-client/get-started/linux/

---

不止是跑这个 `undetected_chromedriver` 这么简单，还要一个浏览器安装在系统里，不然 `api` 启动不了

**Reply by linweiyuan** [reply]
如果远程可以的话，好像没看到类似的 `API`，不知道怎么把它们连起来

而如果 `undetected_chromedriver` 和 `google-chrome` 不在同一个机器上，无法完成初始化

![2023-04-19_22-42](https://user-images.githubusercontent.com/49076004/233111523-a3df1186-65fe-49d0-9b30-87276978ee65.png)

**Reply by linweiyuan** [reply]
现在 `undetected_chromedriver` 就是远程启动的

**Reply by linweiyuan** [reply]
刚试过是可以全部远程连接的，包括 `chromedriver` 和 `google-chrome`，但是不明白方便好多在哪
![2023-04-20_01-09](https://user-images.githubusercontent.com/49076004/233149842-e5b95812-62e6-4a6c-95e7-e25d397fa622.png)

**Reply by linweiyuan** [reply]
如果是这个需求，是不是可以直接在高配电脑上跑 `chatgpt-proxy-server`，然后改 `CHATGPT_PROXY_SERVER` 指向它

**Reply by nephen** [reply]
境外小机器跑个[socks5代理](https://github.com/nephen/go-socks-proxy)就好了，其他的都跑国内

**Comment by linweiyuan** [comment]
> 代理机器，配置很低，搭建docker环境不现实，希望提供编译好的文件，我直接部署

网友总结 `Windows` 版本无 `docker` 部署流程，供参考：https://github.com/linweiyuan/go-chatgpt-api/issues/66
